### ISSUE TITLE
Set Cell Meta Based On Physical Indexes

### ISSUE THREAD

**Issue by aarondfrancis** [handsontable/handsontable#7401]
👋  I've got another request for y'all!

I would love to be able to set cell meta based on physical indexes instead of visual ones.

Of course y'all have the method [`setCellMeta`](https://github.com/handsontable/handsontable/blob/master/src/core.js#L2678), which operates on visual indexes, but since we're primarily working with physical indexes it would be awesome to be able to set meta that way as well. 

I've tried using the `cells` method in the user settings, but have run into two problems. 

1) It runs _all_ the time, way more than I need it to.
2) It's incredibly hard to _unset_ values. If on the first call the method returns `{bar: "baz"}` and later it returns `{}`, then the cell meta object will still contain `{bar: "baz"}` because the old value gets `extend`ed into the meta.

I also previously used `beforeGetCellMeta` and that worked, but it also ran way more often than I needed it to.

This is by no means a pressing feature, but it would be a nice quality of life improvement.

**Comment by AMBudnik** [comment]
Hi @aarondfrancis 

we also have a method that can translate the index [`toPhysicalRow`](https://handsontable.com/docs/8.2.0/Core.html#toPhysicalRow) so if you'd wish to use `setCellMeta` that would be https://jsfiddle.net/AMBudnik/ukw5v7zL/

**Comment by aarondfrancis** [comment]
@AMBudnik thanks for the response! Maybe I'm misunderstanding what you've done here. As far as I can tell, the `setCellMeta` method accepts _visual_ rows and converts them to physical internally.

So you've taken row `0`, converted it to a physical index, passed it into the method, which itself converts it to a physical index _again_. I think your example might work by coincidence. Maybe you meant `toVisualRow` instead?

See: https://jsfiddle.net/zw9mgxj1/1.

Regardless, you can't use `toPhysicalRow` on rows that don't have a visual equivalent, because their visual index will be reported as `null`, so I'm left wanting to use physical indexes again.

**Comment by AMBudnik** [comment]
> Regardless, you can't use toPhysicalRow on rows that don't have a visual equivalent, because their visual index will be reported as null, so I'm left wanting to use physical indexes again.

Yes, that's true. There's definitely a gap here for improvement here :)

**Comment by sl01k** [comment]
We're moving feature requests to our [community forum](https://forum.handsontable.com/) where they can be properly tracked and voted on by the community.

Closing this issue. If this feature is still important to you, please repost it on the forum — your input helps us prioritize.

Thank you!
